lib/models/task.dart

```diff
@@ -203,11 +207,52 @@ class Task {
   String toString() {
     return "Task($name, $id, $recurrenceRules)";
   }
+  
+  String toDetailedString() {
+    return "Task($name, $id, $description, $completed, $location, $color, $recurrenceRules, $tags, $timeStart, $timeDue, $timeCurrent, $timeCreated, $timeModified)";
+  }
 
   @override
   bool operator ==(Object other) {
     if (identical(this, other)) return true;
 
+    if (other is Task &&
+        _name == other._name &&
+        _id == other._id &&
+        _description == other._description &&
+        _completed == other._completed &&
+        _location == other._location &&
+        _color == other._color) {
+          // print("Strings are all equal");
+        } else {
+          print("Strings are not equal");
+        }
+
+    if (other is Task &&
+        _timeStart == other._timeStart &&
+        _timeDue == other._timeDue &&
+        _timeCurrent == other._timeCurrent &&
+        _timeCreated == other._timeCreated &&
+        _timeModified == other._timeModified) {
+          // print("Times are all equal");
+        } else {
+          print("Times are not equal");
+        }
+
+    if (other is Task &&
+        listEquals(_tags, other._tags)) {
+          // print("Tags are equal");
+        } else{
+          print("Tags are not equal");
+        }
+
+    if (other is Task &&
+        _recurrenceRules == other._recurrenceRules) {
+          // print("Recurrence rules are equal");
+        } else {
+          print("Recurrence rules are not equal");
+        }
+
```

Comment:
Is this part necessary?

Reply:
It's more for debugging, but it doesn't impact the final return value. I had a failing test case and I wasn't sure which of the fields weren't equal so I created it. I think it's good to keep in case future code also needs debugging.

Reply:
I think flutter has something called kIsDebug or something to help toggle debug mode.

Reply:
At least it's something my IDE prompts me about

Reply:
I could just comment it out for now so it doesnt print on any comparison?

Reply:
sure